src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -474,20 +473,23 @@ private void RaiseFunctionMetricEvents()
 
                 // We only need to raise events here for functions that aren't completed.
                 // Events are raised immediately for completed functions elsewhere.
-                var runningFunctions = new List<FunctionStartedEvent>();
+                FunctionStartedEvent[] runningFunctionsSnapshot = null;
                 lock (_runningFunctionsSyncLock)
                 {
                     // effectively we're pruning all the completed invocations here
-                    runningFunctions = _runningFunctions = _runningFunctions.Where(p => !p.Completed).ToList();
```

Comment:
The cause of the issue was here - the runningFunctions local was intended to be a snapshot not a reference to _runningFunctions. Once outside the lock, _runningFunctions can be modified on other threads as functions are executed, which can cause failures below while we're enumerating. This results in "collection was modified" exceptions that caused the activity task to fail silently (background task exception).